### PR TITLE
[IMP] hr: allow employee public users to update profile picture

### PR DESCRIPTION
**Pull request by sinachaichi**
Current restrictions prevent public users from updating their profile pictures. The issue stems from the onchange method fetching fields that the user group lacks access to. By overriding the onchange method we filter out inaccessible fields before calling the super method. This ensures that public users can update their profile pictures without encountering access errors.

Task-3582195

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr/models/res_users.py [hunk 1]
```diff
@@ -334,3 +334,7 @@ def action_open_employees(self):
             'res_id': employees.id,
             'view_mode': 'form',
         }
+
+    def onchange(self, values, field_names, fields_spec):
+        fields_spec = {key: fields_spec[key] for key in self._get_employee_fields_to_sync() if key in fields_spec}
+        return super().onchange(values, field_names, fields_spec)
```
Comment (sofiagvaladze) [inline addons/hr/models/res_users.py:340]: Prpbably we should only filter out in specific cases and not always
```suggestion
    def onchange(self, values, field_names, fields_spec):
        # Some of the fields are only allowed for groups Employees / Officer
        # we can avoid building the snapshots for those fields
        fields_to_read = self._get_employee_fields_to_sync()
        if any(field in field_names for field in fields_to_read):
            values = {key: val for key, val in values.items() if key in fields_to_read}
            fields_spec = {key: val for key, val in fields_spec.items() if key in fields_to_read}
        return super().onchange(values, field_names, fields_spec)
```

Comment (cmsi-odoo) [reply addons/hr/models/res_users.py:340]: Yes, I saw the tests failed and was trying to fix it.

File: addons/hr/models/res_users.py [hunk 2]
```diff
@@ -334,3 +334,12 @@ def action_open_employees(self):
             'res_id': employees.id,
             'view_mode': 'form',
         }
+
+    def onchange(self, values, field_names, fields_spec):
```
Comment (tivisse) [inline addons/hr/models/res_users.py:338]: Overriding ORM methods for a simple use case is usually a bad idea. 

Did you try adding the field ``image_1920`` into the ``HR_WRITABLE_FIELDS`` instead ? 

Normally it should solve your use case.

Comment (sofiagvaladze) [reply addons/hr/models/res_users.py:338]: Adding `image_1920` into `HR_WRITABLE_FIELDS` and `HR_READABLE_FIELDS` does not do a trick ...

We get seemingly unrelated error when we try to change user picture:

```
Access Error

The requested operation can not be completed due to security restrictions.

Document type: Employee (hr.employee)
Operation: read
User: 6
Fields:
- private_street (allowed for groups 'Employees / Officer: Manage all employees')
```

My guess is that this is due to the changes in `onchage`. When we change something in the form view, it reads all the fields that are on the form
https://github.com/odoo/odoo/blob/40934f4089c82f76ca88173271eb10dad7309e1a/addons/web/models/models.py#L889-L898

It reads them here https://github.com/odoo/odoo/blob/40934f4089c82f76ca88173271eb10dad7309e1a/addons/web/models/models.py#L1004

I guess with that change, we should have overriden  onchange, the same way do read method? But looking back at the change we should probably have `fields_to_read = HR_READABLE_FIELDS + HR_WRITABLE_FIELDS`

**Comment by rco-odoo** [comment]
Suggestion: start by writing a test case that reproduces the issue.
You probably need to use a `Form` object to trigger it.
Once you have it, you can use that test code for debugging and finding the right way to fix the issue.

**Comment by rco-odoo** [comment]
A better fix was found in https://github.com/odoo/odoo/pull/148997.

**Pull request closed without merging**